api/seedmanager/seedmanager.go

```diff
@@ -53,25 +53,33 @@ func New() *SeedManager {
 
 // NewSpecified creates new seed manager with custom params
 func NewSpecified(ttl time.Duration, cleanPeriod time.Duration) *SeedManager {
-	sm := SeedManager{seedPool: make(map[Seed]storedSeed), ttl: ttl, stopped: false}
+	sm := SeedManager{
+		seedPool: make(map[Seed]storedSeed),
+		ttl:      ttl,
+		stopped:  make(chan struct{}),
+	}
+
+	ticker := time.NewTicker(cleanPeriod)
+
 	go func() {
-		for range time.Tick(cleanPeriod) {
-			sm.deleteExpired()
-			sm.mutex.RLock()
-			if sm.stopped {
-				break
+		var stop = false
+		for !stop {
+			select {
+			case <-ticker.C:
+				sm.deleteExpired()
+			case <-sm.stopped:
+				stop = true
 			}
-			sm.mutex.RUnlock()
 		}
+		sm.stopped <- struct{}{}
 	}()
 
 	return &sm
 }
 
 func (sm *SeedManager) Stop() {
```

Comment:
Should we add call to `sm.Stop()` in `Runner.Stop()` func?

Reply:
why not